Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -6,13 +6,81 @@ import Common
 import ComponentLibrary
 import UIKit
 import Shared
+import Redux
+
+struct FakespotState: ScreenState, Equatable {
+    var isSettingsExpanded: Bool
+    var isReviewQualityExpanded: Bool
+    let logger: Logger = DefaultLogger.shared
+
+    enum Action: Redux.Action {
+        case settingsStateDidChange
+        case reviewQualityDidChange
+        case urlDidChange
+    }
+
+    init() {
+        self.isSettingsExpanded = false
+        self.isReviewQualityExpanded = false
+    }
+
+    init(isSettingsExpanded: Bool, isReviewQualityExpanded: Bool) {
+        self.isSettingsExpanded = isSettingsExpanded
+        self.isReviewQualityExpanded = isReviewQualityExpanded
+    }
+
+    init(_ appState: AppState) {
+        guard let state = store.state.screenState(FakespotState.self, for: .fakespot) else {
+            self.init(isSettingsExpanded: false, isReviewQualityExpanded: false)
+            logger.log("Error retrieving screen state",
```

Comment:
I've seen this pattern in the theme state, and adding logger sadly it breaks the Equality. I see logger as a dependency and I think we should keep the state in it's most pure form. and not couple the store with the state.

Reply:
I see your point, I added logger later in the implementation but there is no strong reason to keep it